### ISSUE TITLE
Map property by using custom named mapping method

### ISSUE THREAD

**Issue by Yutsa** [riok/mapperly#783]
**Is your feature request related to a problem? Please describe.**

I want to map an object with some custom logic. In this mapping I would want to say map this int to this string using this method, and this int to this other string using this other method.

Currently the only way to do that is to use `After` mapping, and it works great. However I want to use these custom mappings inside the `IQueryable` mapping. Currently the generated method for `IQueryable` can't retrieve theses custom mappings because it only retrieves custom mappings from type A to type B and there is no way to specify a mapping using a particual method.

**Describe the solution you'd like**

```csharp
    [MapProperty(nameof(WorkHeader.Works), nameof(WorkHeaderDto.Comment), nameof(CustomMappingMethod))]
    [MapProperty(nameof(WorkHeader.Works), nameof(WorkHeaderDto.TypeTravail), nameof(OtherCustomMapping))]
    private static partial WorkHeaderDto MapInternal(this WorkHeader workHeader);

    private static string CustomMappingMethod(ICollection<Work> works)
    {
        // Custom logic to return a new string
        return "";
    }
    
    private static string OtherCustomMapping(ICollection<Work> works)
    {
        // Custom logic to return a new string
        return "";
    }
```

This way Mapperly could pick up this mapping and inline the method call inside the `Select` call from the generated `IQueryable` mapping.

As you can see I have two mappings from `ICollection<Work>` to `string` that I want to use for the same source but with different targets.

**Describe alternatives you've considered**

I've tried using `After` mapping but since calling a method violates Expression rules, it won't be transposed into the SQL query.

Basically, I'm asking for this feature because I feel it is the only way to apply this type of custom mapping logic to the `IQueryable` mapping. But if there is another way to get both auto mapping + explicit custom mapping inside the `IQueryable` mapping, I'm open to it

**Comment by latonz** [comment]
Rel. https://github.com/riok/mapperly/discussions/162

**Comment by Yutsa** [comment]
Yep, a `UseMethod` property in  `MapPropertyAttribute` would be great, it would solve the issues that prevent us from using Mapperly in our production app.

**Comment by latonz** [comment]
@Yutsa fair point. Do you have a proposal on how this could be achieved? Probably a `IgnoreMappingMethodAttribute`?

**Comment by latonz** [comment]
I updated my proposal to name the property `UseMapping` instead of `UseMethod`. External mappings could then referenced by their simple names by applying the `NamedMappingAttribute` (see https://github.com/riok/mapperly/issues/513).

**Comment by UltraWelfare** [comment]
I feel like this is a needed feature that would allow for more customization in mapperly.
I stumbled upon [the docs about user implemented methods](https://mapperly.riok.app/docs/configuration/user-implemented-methods/) but it sadly only supports types.

In an even easier example on why this is needed and is a great addition:
```csharp
[Mapper]
public partial class CarMapper
{
    public partial CarDto CarToCarDto(Car car);

    private int TimeSpanToHours(TimeSpan t) => t.Hours;
}
```

What if the mapper has two TimeSpans that one needs to be converted to hours and the other to minutes?

I'm all-in for what @Yutsa suggested, something like this:

```csharp
[MapProperty(nameof(Car.T1), nameof(CarDto.Hours), nameof(ConvertToHours))]    
[MapProperty(nameof(Car.T2), nameof(CarDto.Minutes), nameof(ConvertToMinutes))]
```

I don't understand how #891 is a proposal for this feature.

**Comment by latonz** [comment]
The inlining part is handled in https://github.com/riok/mapperly/issues/783.

**Comment by latonz** [comment]
This is released on the next channel as version `3.5.0-next.2`. Any feedback would be appreciated 😊

**Comment by lfpraca** [comment]
Is this a possible solution for the usage in #513 ? As in, if I have a mapper for a nested class in another mapper class, does this allow me to specify that as a custom mapper so that it can map the class that has this nested class without having to specify both mappers in the same file?
